metadata-service/plugin/src/main/java/com/datahub/plugins/loader/IsolatedClassLoaderImpl.java

```diff
@@ -0,0 +1,294 @@
+          pluginToLoad.getName()));
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void createJarEntryMap() throws IOException {
+    this._pluginJarRef = new JarFile(this._pluginConfig.getPluginJarPath().toFile());
+    this._classPathVsZipEntry = new HashMap<>();
+    for (Enumeration<JarEntry> enums = this._pluginJarRef.entries(); enums.hasMoreElements(); ) {
+      JarEntry entry = enums.nextElement();
+      if (entry.getName().endsWith("/")) {
+        // we don't want to keep directories in map
+        continue;
+      }
+      this._classPathVsZipEntry.put(entry.getName(), entry);
+    }
+  }
+
+  /**
+   * Load plugin class from jar given in pluginToLoad parameter and return instance of class which implements Plugin interface.
+   * This method verifies whether loaded plugin is assignable to expectedInstanceOf class
+   * @param expectedInstanceOf class instance of interface caller is expecting
+   * @return Instance of Plugin
+   * @throws ClassNotFoundException className parameter available in Plugin configuration is not found
+   */
+  @Nonnull
+  public Plugin instantiatePlugin(Class<? extends Plugin> expectedInstanceOf) throws ClassNotFoundException {
+    Class<?> clazz = this.loadClass(this._pluginConfig.getClassName(), true);
+
+    try {
+      log.debug("Creating instance of plugin {}", this._pluginConfig.getClassName());
+      Plugin plugin = (Plugin) clazz.newInstance();
+      // Check loaded plugin has implemented the proper implementation of child interface
+      if (!expectedInstanceOf.isAssignableFrom(clazz)) {
+        throw new InstantiationException(
+            String.format("In plugin %s, the class %s has not implemented the interface %s",
+                this._pluginConfig.getName(), plugin.getClass().getCanonicalName(),
+                expectedInstanceOf.getCanonicalName()));
+      }
+      log.debug("Successfully created instance of plugin {}", this._pluginConfig.getClassName());
+      return plugin;
+    } catch (InstantiationException | IllegalAccessException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private String classNameToPath(String resourceName) {
+    // in-case of java class , we need to append the .class to last element
+    return resourceName.replaceAll("\\.", "/") + ".class";
+  }
+
+  private byte[] getClassData(ZipEntry zipEntry) {
+    try (InputStream ins = this._pluginJarRef.getInputStream(zipEntry);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      byte[] buffer = new byte[4096];
+      int bytesNumRead;
+      while ((bytesNumRead = ins.read(buffer)) != -1) {
+        baos.write(buffer, 0, bytesNumRead);
+      }
+      return baos.toByteArray();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return new byte[]{};
+  }
+
+  @Override
+  protected Class<?> loadClass(String s, boolean b) throws ClassNotFoundException {
+    log.debug(String.format("Load class %s", s));
```

Comment:
Why these debug files? In general we need more logging.

Reply:
Added more logging